sky/server/common.py

```diff
@@ -213,8 +217,13 @@ def get_dashboard_url(server_url: str,
 
 
 @annotations.lru_cache(scope='global')
-def is_api_server_local():
-    return get_server_url() in AVAILABLE_LOCAL_API_SERVER_URLS
+def is_api_server_local(host: Optional[str] = DEFAULT_SERVER_HOST):
+    server_url = get_server_url(host)
+    if '://' in server_url:
+        server_host = server_url.split('://')[1].split(':')[0]
+    else:
+        server_host = server_url.split(':')[0]
+    return server_host in AVAILBLE_LOCAL_API_SERVER_HOSTS
```

Comment:
Could we use a standard parser here instead of trying to parse it ourselves? Could have some issues, e.g. for this server URL `http://localhost:password@remote.server.com`